test/test.apirequest.ts

```diff
@@ -298,6 +298,31 @@ describe('createAPIRequest', () => {
       });
       scope.done();
     });
+
+    it('should rewrite url to match default rootUrl', async () => {
+      const rootUrl = 'http://www.googleapis.com/';
+      const path = '/api/service';
+      const normalize = (x: string) => x.replace(/([^:]\/)\/+/g, '$1');
+      const scope = nock(rootUrl)
+        .get(path)
+        .reply(200);
+      const res = await createAPIRequest<FakeParams>({
+        options: {
+          url: normalize('https://www.googleapis.com/' + path),
+        },
+        params: {},
+        requiredParams: [],
+        pathParams: [],
+        context: {
+          _options: {
+            rootUrl,
+          },
+        },
+      });
+      scope.done();
+      const expectedUrl = normalize(rootUrl + path);
+      assert.strictEqual(res.config.url, expectedUrl);
```

Comment:
since part of what we want to test is the `url.resolve` logic, I'd probably make `expectedUrl` a `string`, rather than calling the `normalize` method on it :+1: